Add Ndiff Wrapper

juicebyjustin opened the pull request:
Hi Thom, another pull request for you. I added an Ndiff wrapper.

I included 2 XML Nmap output files for testing. Honestly, I don't know where test files should go so I placed them in a folder inside bin/debug in the test project. Let me know how you want it and I will modify to fit.

I'm using VS2017 so I don't know if the commited .csproj files will cause any issue.

Commit:
Add Ndiff Wrapper

Hi Thom, another pull request for you. I added an Ndiff wrapper

I included 2 XML Nmap output files for testing. Honestly, I don't know where test files should go so I placed them in a folder inside bin/debug folder in the test project. Let me know how you want it and I will modify to fit.

I'm using VS2017 so I don't know if the commited .csproj files will cause any issue.

thomdixon commented:
Thanks again! I'll have to check this one out later this week (probably this weekend). I don't have access to a computer at the moment, so I can't check if the csproj file causes any issues. The code looks good. I'd like to repeat the `NmapOptions` pattern for Ndiff, though.

thomdixon commented on .gitignore:
```diff
@@ -0,0 +1,10 @@
+
```
Can you maybe add a `.gitignore` from [gitignore.io](https://www.gitignore.io/api/csharp%2Cvisualstudio) instead?

juicebyjustin replied:
Will do.

thomdixon commented on SaltwaterTaffy/Ndiff.cs:
```diff
@@ -0,0 +1,57 @@
+        /// <param name="verbose">Also show hosts and ports that haven't changed.</param>
+        /// <param name="xml">Display output in XML format.</param>
+        /// <returns>Output from Ndiff application.</returns>
+        public static string RunDiff(bool verbose, bool xml, string file1, string file2)
```
Would prefer to use the `NmapOptions` pattern here (maybe an `NdiffOptions` object), but I'm fine leaving it like this, if this is the only functionality you need.

juicebyjustin replied:
That's make sense to use the NmapOptions pattern. I'll try to have another commit before the weekend with this and the other changes per your comments.

thomdixon commented on SaltwaterTaffy/Ndiff.cs:
```diff
@@ -0,0 +1,57 @@
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(string.Format("Failed to run diff. Ex: {0}", ex.ToString()));
```
Would use [`NmapException`](https://github.com/thomdixon/SaltwaterTaffy/blob/master/SaltwaterTaffy/Nmap.cs#L505) here instead.

juicebyjustin replied:
Certainly.

thomdixon commented on SaltwaterTaffy/Ndiff.cs:
```diff
@@ -0,0 +1,57 @@
+        /// <param name="verbose">Also show hosts and ports that haven't changed.</param>
+        /// <param name="xml">Display output in XML format.</param>
+        /// <returns>Output from Ndiff application.</returns>
+        public static string RunDiff(bool verbose, bool xml, string file1, string file2)
```
Why not name this `Run`?

juicebyjustin replied:
I honestly do not know. I'll change to Run.

thomdixon reviewed (changes requested):
I have a few comments that I'd like your input on, but otherwise looks good to me.

Commit:
Made Changes Based on Comments

Changes were made based on your comments. The NdiffOptions was easier than I thought to implement.

jmooney5115 commented:
Hi. The changes were easier than expected. I'm new to pull requests. How do add the modified code here? I committed it to the same 'ndiff-addition' branch.

thomdixon reviewed (approved):
Looks great! Thanks so much!

juicebyjustin commented:
You're welcome, I'm glad to help.
